homeassistant/components/cover/mqtt.py

```diff
@@ -67,30 +95,45 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
         config.get(CONF_PAYLOAD_STOP),
         config.get(CONF_OPTIMISTIC),
         value_template,
+        config.get(CONF_TILT_OPEN_POSITION),
+        config.get(CONF_TILT_CLOSED_POSITION),
+        config.get(CONF_TILT_MIN),
+        config.get(CONF_TILT_MAX),
+        config.get(CONF_TILT_STATE_OPTIMISTIC),
     )])
 
 
 class MqttCover(CoverDevice):
     """Representation of a cover that can be controlled using MQTT."""
 
-    def __init__(self, name, state_topic, command_topic, qos, retain,
-                 state_open, state_closed, payload_open, payload_close,
-                 payload_stop, optimistic, value_template):
+    def __init__(self, name, state_topic, command_topic, tilt_command_topic,
+                 tilt_status_topic, qos, retain, state_open, state_closed,
+                 payload_open, payload_close, payload_stop,
+                 optimistic, value_template, tilt_open_position,
+                 tilt_closed_position, tilt_min, tilt_max, tilt_optimistic):
         """Initialize the cover."""
         self._position = None
         self._state = None
         self._name = name
         self._state_topic = state_topic
         self._command_topic = command_topic
+        self._tilt_command_topic = tilt_command_topic
+        self._tilt_status_topic = tilt_status_topic
         self._qos = qos
         self._payload_open = payload_open
         self._payload_close = payload_close
         self._payload_stop = payload_stop
         self._state_open = state_open
         self._state_closed = state_closed
         self._retain = retain
+        self._tilt_open_position = tilt_open_position
+        self._tilt_closed_position = tilt_closed_position
         self._optimistic = optimistic or state_topic is None
         self._template = value_template
+        self._tilt_value = STATE_UNKNOWN
```

Comment:
This one still needs to be updated to `None`. If we add `supported_features` it should be OK. What we don't want to do is show tilt controls for all existing MQTT covers that don't have tilt configured. If we add supported features as shown below it should work. You could also define the groups of features as constants at the top of the module so they don't have to be recomputed each time. (See [hue](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/light/hue.py#L49) for an example).

```python
@property
def supported_features(self):
    supported_features = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP

    if self._tilt_command_topic is not None:
        supported_features |= (
            SUPPORT_OPEN_TILT | SUPPORT_CLOSE_TILT | SUPPORT_STOP_TILT |
            SUPPORT_SET_TILT_POSITION)

    return supported_features